registry/storage/driver/gcs/gcs.go

```diff
@@ -659,12 +641,17 @@ func (d *driver) List(ctx context.Context, path string) ([]string, error) {
 		Delimiter: "/",
 		Prefix:    d.pathToDirKey(path),
 	}
+	objects := d.bucket.Objects(ctx, query)
+
 	list := make([]string, 0, 64)
-	objects, err := storageListObjects(ctx, d.bucket, query, d.gcs)
-	if err != nil {
-		return nil, err
-	}
-	for _, object := range objects {
+	for {
+		object, err := objects.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Also see my other comment; we should be consistent, and pick one approach for these;

```go
if err != nil {
	if err == iterator.Done {
		break
	}
	return nil, err
}
```